gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -197,6 +224,29 @@ protected Path getPublisherOutputDir(WorkUnitState workUnitState, int branchId)
     return WriterUtils.getDataPublisherFinalDir(workUnitState, this.numBranches, branchId);
   }
 
+  protected void addSingleTaskWriterOutputToExistingDir(Path writerOutputDir, Path publisherOutputDir,
+      WorkUnitState workUnitState, int branchId, ParallelRunner parallelRunner) throws IOException {
+    Preconditions.checkArgument(workUnitState.contains(ConfigurationKeys.WRITER_FINAL_OUTPUT_PATH), "Missing property "
+        + ConfigurationKeys.WRITER_FINAL_OUTPUT_PATH + ", which is needed for publishing data of a single task");
+
+    Iterable<String> taskOutputFiles = workUnitState.getPropAsList(ConfigurationKeys.WRITER_FINAL_OUTPUT_PATH);
```

Comment:
Sounds like the config property should be named `WRITER_FINAL_OUTPUT_FILE_PATHS`.